File: superstitioApi/java/superstitioapi/utils/ImgUtility.java
package superstitioapi.utils;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;

public class ImgUtility {


    public final static Color nullColor = new Color(0, 0, 0, 0);

    public static void draw(final SpriteBatch sb, Texture texture, float x, float y,
                            float width, float height, float rotation) {
        sb.draw(texture, x, y, 0, 0, width, height, 1.0f, 1.0f, rotation, 0, 0, texture.getWidth(), texture.getHeight(),
                false, false);
    }

    public static void draw(final SpriteBatch sb, Texture texture, float x, float y,
                            float width, float height, float rotation, boolean flipX, boolean flipY) {
        sb.draw(texture, x, y, 0, 0, width, height, 1.0f, 1.0f, rotation, 0, 0, texture.getWidth(), texture.getHeight(),
                flipX, flipY);
    }

    public static Color mixColor(Color colorBase, Color colorInsert, float InsertRate) {
        float r = (colorBase.r + colorInsert.r * InsertRate) / (1.0f + InsertRate);
        float g = (colorBase.g + colorInsert.g * InsertRate) / (1.0f + InsertRate);
        float b = (colorBase.b + colorInsert.b * InsertRate) / (1.0f + InsertRate);
        float a = (colorBase.a + colorInsert.a * InsertRate) / (1.0f + InsertRate);
        return new Color(r, g, b, a);
    }

    public static Color mixColor(Color colorBase, Color colorInsert, float InsertRate, float alpha) {
        float r = (colorBase.r + colorInsert.r * InsertRate) / (1.0f + InsertRate);
        float g = (colorBase.g + colorInsert.g * InsertRate) / (1.0f + InsertRate);
        float b = (colorBase.b + colorInsert.b * InsertRate) / (1.0f + InsertRate);
        return new Color(r, g, b, alpha);
    }


}

File: superstitioApi/java/superstitioapi/utils/ImgUtility.kt
package superstitioapi.utils

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.SpriteBatch

object ImgUtility
{
    val nullColor: Color = Color(0f, 0f, 0f, 0f)

    fun draw(
        sb: SpriteBatch, texture: Texture, x: Float, y: Float,
        width: Float, height: Float, rotation: Float
    )
    {
        sb.draw(
            texture, x, y, 0f, 0f, width, height, 1.0f, 1.0f, rotation, 0, 0, texture.width, texture.height,
            false, false
        )
    }

    fun draw(
        sb: SpriteBatch, texture: Texture, x: Float, y: Float,
        width: Float, height: Float, rotation: Float, flipX: Boolean, flipY: Boolean
    )
    {
        sb.draw(
            texture, x, y, 0f, 0f, width, height, 1.0f, 1.0f, rotation, 0, 0, texture.width, texture.height,
            flipX, flipY
        )
    }

    fun mixColor(colorBase: Color, colorInsert: Color, InsertRate: Float): Color
    {
        val r = (colorBase.r + colorInsert.r * InsertRate) / (1.0f + InsertRate)
        val g = (colorBase.g + colorInsert.g * InsertRate) / (1.0f + InsertRate)
        val b = (colorBase.b + colorInsert.b * InsertRate) / (1.0f + InsertRate)
        val a = (colorBase.a + colorInsert.a * InsertRate) / (1.0f + InsertRate)
        return Color(r, g, b, a)
    }

    fun mixColor(colorBase: Color, colorInsert: Color, InsertRate: Float, alpha: Float): Color
    {
        val r = (colorBase.r + colorInsert.r * InsertRate) / (1.0f + InsertRate)
        val g = (colorBase.g + colorInsert.g * InsertRate) / (1.0f + InsertRate)
        val b = (colorBase.b + colorInsert.b * InsertRate) / (1.0f + InsertRate)
        return Color(r, g, b, alpha)
    }
}
